alyssa_mckee/SQL-Flask/Friends/server.py

```diff
@@ -0,0 +1,59 @@
+from flask import Flask, render_template, redirect, request, flash
+from mysqlconnection import MySQLConnector 
+import re
+import sys
+
+def log(obj):
+	print(obj, file=sys.stderr)
+
+NAME_REGEX = re.compile(r'^[a-zA-Z-]+$')
+
+app = Flask(__name__)
+app.secret_key="Mo!W6s1nG@e545$9j9OnEpyuBK*&M@X*l1HBORF3&I"
+
+mysql = MySQLConnector(app, 'facelessbook_db') 
+
+@app.route("/")
+def index():
+	query = '''SELECT CONCAT_WS(' ', first_name, last_name) AS name, age, DATE_FORMAT(created_at, "%M %e") AS date, YEAR(created_at) AS year FROM friends'''
+	log(mysql.query_db(query))
+	data = mysql.query_db(query)
+	return render_template("index.html", data=data)
+
+@app.route('/process', methods=["POST"])
+def process():
```

Comment:
This method's functionality looks great. Try practicing moving some functionality into multiple functions to keep our methods clear and concise. A separate validation function seems like an easy way to make this method more discernible.